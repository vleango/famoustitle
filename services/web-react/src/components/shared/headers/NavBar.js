import React from 'react';
import { Link } from 'react-router-dom';

export class NavBar extends React.Component {

    render() {
        return (
            <ul className="nav navbar sticky-top navbar-light bg-light">
                <li className="nav-item">
                    <Link className="nav-link active nav-title" to={`/`}>FamousTitle</Link>
                </li>
            </ul>
        );
    }
}
