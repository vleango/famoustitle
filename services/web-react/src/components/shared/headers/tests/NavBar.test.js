import React from 'react';
import { shallow } from 'enzyme';
import { NavBar } from '../NavBar';

let wrapper;

describe('Shared', () => {
    describe('NavBar', () => {
        wrapper = shallow(<NavBar />);

        it('should correctly render NavBar', () => {
            expect(wrapper).toMatchSnapshot();
        });
    });
});
