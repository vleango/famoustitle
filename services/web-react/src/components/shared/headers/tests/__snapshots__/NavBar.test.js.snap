// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shared NavBar should correctly render NavBar 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NavBar />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <li
        className="nav-item"
>
        <Link
                className="nav-link active nav-title"
                replace={false}
                to="/"
        >
                FamousTitle
        </Link>
</li>,
      "className": "nav navbar sticky-top navbar-light bg-light",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Link
          className="nav-link active nav-title"
          replace={false}
          to="/"
>
          FamousTitle
</Link>,
        "className": "nav-item",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "FamousTitle",
          "className": "nav-link active nav-title",
          "replace": false,
          "to": "/",
        },
        "ref": null,
        "rendered": "FamousTitle",
        "type": [Function],
      },
      "type": "li",
    },
    "type": "ul",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <li
          className="nav-item"
>
          <Link
                    className="nav-link active nav-title"
                    replace={false}
                    to="/"
          >
                    FamousTitle
          </Link>
</li>,
        "className": "nav navbar sticky-top navbar-light bg-light",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Link
            className="nav-link active nav-title"
            replace={false}
            to="/"
>
            FamousTitle
</Link>,
          "className": "nav-item",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "FamousTitle",
            "className": "nav-link active nav-title",
            "replace": false,
            "to": "/",
          },
          "ref": null,
          "rendered": "FamousTitle",
          "type": [Function],
        },
        "type": "li",
      },
      "type": "ul",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
